### PR TITLE
[nrfconnect] Enable commissionable node advertising

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
Thread platforms such as nRF Connect SDK don't support commissionable node advertising.

#### Change overview
1. Clean up DNS-SD default configuration as different settings were defined in two different places.
3. Clean up conditions for enabling commissionable node advertising.
4. Enable commissionable node advertising for nRF Connect platform.

#### Testing
Tested using nRF Connect Lock example:
1. Checked that after pressing Button 3 the device advertises both operational and commissionable node services.
2. Likewise, checked that after commissioning with Python CHIP Controller the device advertises both operational and commissionable node services.
3. Verified that commissionable node services are searchable by subtypes, i.e. commands like `avahi-browse -rt _S0._sub._matterc._udp` return correct data.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: src/app/server/RendezvousServer.cpp
Comment (bzbarsky-apple) [inline src/app/server/RendezvousServer.cpp:66]: This change seems likely to break operational advertisement in various cases, due to https://github.com/project-chip/connectedhomeip/issues/7911

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by woody-apple — DISMISSED** [review]
It looks like this changes logic that is shared amongst multiple platforms. What unit tests cover this shared code? It seems like we should be adding new unit tests here...

File: src/app/server/RendezvousServer.cpp
Comment (Damian-Nordic) [reply src/app/server/RendezvousServer.cpp:66]: Meh. Sounds like I need to add more ifdefs.

**Comment by Damian-Nordic** [comment]
> It looks like this changes logic that is shared amongst multiple platforms. What unit tests cover this shared code? It seems like we should be adding new unit tests here...

Which logic do you mean? We probably can't unit-test the OpenThread changes as they are only used by platforms that are run on hardware.

**Review by LuDuda — APPROVED** [review]

File: src/app/server/RendezvousServer.cpp
Comment (Damian-Nordic) [reply src/app/server/RendezvousServer.cpp]: Ok, reverted that change, but also changed conditions in `server/Mdns.cpp` as we discussed on slack.

**Comment by Damian-Nordic** [comment]
> It looks like this changes logic that is shared amongst multiple platforms. What unit tests cover this shared code? It seems like we should be adding new unit tests here...

I reduced changes in the common code to one condition change in `server/Mdns.cpp` which seems to make sense anyway - we shouldn't advertise operational nodes with any default node IDs. Apparently, some changes to OTBR must be made to be able to test mDNS in the cirque framework - before that I'm not sure how we can test that part automatically. But I'm welcome to any suggestions.

**Comment by kkasperczyk-no** [comment]
@bzbarsky-apple @woody-apple could you re-review and see if there are any other changes to request? I think we are still missing here approval from your side.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
@woody-apple PTAL?

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
> What unit tests cover this shared code?

Right now nothing.  The thing to unit-test here would be what sorts of advertisements we actually produce in various situations, and we don't have very good infrastructure for this right now....

I filed https://github.com/project-chip/connectedhomeip/issues/8769 for tracking the tests getting added, but don't think we should block the PR on them.

**Pull request merged**
